### PR TITLE
allauth: logout when canceling 2fa

### PR DESCRIPTION
**Pull request by stsewd**
Closes https://github.com/readthedocs/ext-theme/issues/501

### CODE REVIEW

File: readthedocsext/theme/templates/mfa/authenticate.html [hunk 1]
```diff
@@ -22,12 +22,22 @@
...
+      <button class="ui button" type="submit" form="logout-from-stage">
+        {% trans "Cancel" %}
+      </button>
       <div class="right menu">
         <button class="ui primary button" type="submit">{% trans "Verify" %}</button>
       </div>
     </div>
   </form>
+
+  <!-- We can't have nested forms, so we need to have a separate form for "Cancel" -->
```
Comment (agjohnson) [inline readthedocsext/theme/templates/mfa/authenticate.html:35]: Avoid using HTML comments unless we really need the output in the final HTML. Template comments should be used instead.

```suggestion
  {# We can't have nested forms, so we need to have a separate form for "Cancel" #}                                                                    
```

**Review by agjohnson — APPROVED** [review]
This looks good at a code level, but can you post a screen grab of the UI to verify?

**Comment by stsewd** [comment]
### before

Screenshot from 2024-10-31 12-59-13

### after

Screenshot from 2024-10-31 12-58-53

The outline changed

File: readthedocsext/theme/templates/mfa/authenticate.html [hunk 2]
```diff
@@ -22,12 +22,22 @@
     {% csrf_token %}
     {{ form|crispy }}
     {{ redirect_field }}
-
     <div class="ui stackable relaxed text menu">
-      <a class="item" href="{% url "account_login" %}">{% trans "Cancel" %}</a>
+      <button class="ui button" type="submit" form="logout-from-stage">
+        {% trans "Cancel" %}
+      </button>
```
Comment (agjohnson) [inline readthedocsext/theme/templates/mfa/authenticate.html:28]: For now, I'd like to retain that any time we use a left link like this, it _is not a button styling_. I might want to change this in the future, but for now, this has been the pattern on other UIs.

I'd move this button into the `.right.menu`. That would give a button cluster similar to a modal.

**Comment by stsewd** [comment]
Screenshot 2024-11-04 at 14-16-32 Verify sign in - Read the Docs

File: readthedocsext/theme/templates/mfa/authenticate.html [hunk 3]
```diff
@@ -22,7 +22,7 @@
     {% csrf_token %}
     {{ form|crispy }}
     {{ redirect_field }}
-    <div class="ui stackable relaxed text menu">
+    <div class="ui right floated text menu">
```
Comment (agjohnson) [inline readthedocsext/theme/templates/mfa/authenticate.html:25]: This is different than what I was describing, and probably has some negative effects.

You should move the button to the `div.right.menu` directly below the button. A right float menu probably displays poorly on mobile viewports.

**Comment by stsewd** [comment]
Screenshot 2024-11-05 at 10-39-26 Verify sign in - Read the Docs


Screenshot 2024-11-05 at 10-39-16 Verify sign in - Read the Docs

**Review by agjohnson — APPROVED** [review]

**Pull request merged**